src/sentry/integrations/github/client.py

```diff
@@ -45,7 +47,53 @@ def search_repositories(self, query):
         return self.get("/search/repositories", params={"q": query})
 
     def get_assignees(self, repo):
-        return self.get(f"/repos/{repo}/assignees")
+        return self.get_with_pagination(f"/repos/{repo}/assignees")
+
+    def get_with_pagination(self, path, *args, **kwargs):
+        """
+        Github uses the Link header to provide pagination links. Github recommends using the provided link relations and not constructing our own URL.
+        https://docs.github.com/en/rest/guides/traversing-with-pagination
+        """
+        try:
+            with sentry_sdk.configure_scope() as scope:
+                parent_span_id = scope.span.span_id
+                trace_id = scope.span.trace_id
+        except AttributeError:
+            parent_span_id = None
+            trace_id = None
+
+        with sentry_sdk.start_transaction(
```

Comment:
I think we should be using a span for this, I think in most cases we are going to want spans going forward. I know they might not be as easy to search on but if you really need we can add a tag to the `scope`.

Reply:
@MeredithAnya I spoke with @wmak on this and spans are not indexed, so you can't query in Discover. Transactions, on the other hand, are indexed.